flink-tests/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterTriggerSavepointIT.java

```diff
@@ -0,0 +1,219 @@
+ *
+ * @see org.apache.flink.runtime.jobmaster.JobMaster
+ */
+@Category(Flip6.class)
+public class JobMasterTriggerSavepointIT extends AbstractTestBase {
+
+	private static CountDownLatch invokeLatch;
+
+	private static volatile CountDownLatch triggerCheckpointLatch;
+
+	@Rule
+	public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+	private Path savepointDirectory;
+	private MiniClusterClient clusterClient;
+	private JobGraph jobGraph;
+
+	@Before
+	public void setUp() throws Exception {
+		invokeLatch = new CountDownLatch(1);
+		triggerCheckpointLatch = new CountDownLatch(1);
+		savepointDirectory = temporaryFolder.newFolder().toPath();
+
+		Assume.assumeTrue(
+			"ClusterClient is not an instance of MiniClusterClient",
+			miniClusterResource.getClusterClient() instanceof MiniClusterClient);
+
+		clusterClient = (MiniClusterClient) miniClusterResource.getClusterClient();
+		clusterClient.setDetached(true);
+
+		jobGraph = new JobGraph();
+
+		final JobVertex vertex = new JobVertex("testVertex");
+		vertex.setInvokableClass(NoOpBlockingInvokable.class);
+		jobGraph.addVertex(vertex);
+
+		jobGraph.setSnapshotSettings(new JobCheckpointingSettings(
+			Collections.singletonList(vertex.getID()),
+			Collections.singletonList(vertex.getID()),
+			Collections.singletonList(vertex.getID()),
+			new CheckpointCoordinatorConfiguration(
+				10,
+				60_000,
+				10,
+				1,
+				CheckpointRetentionPolicy.NEVER_RETAIN_AFTER_TERMINATION,
+				true),
+			null
+		));
+
+		clusterClient.submitJob(jobGraph, ClassLoader.getSystemClassLoader());
+		invokeLatch.await(60, TimeUnit.SECONDS);
+		waitForJob();
+	}
+
+	@Test
+	public void testStopJobAfterSavepoint() throws Exception {
+		final String savepointLocation = cancelWithSavepoint();
+		final JobStatus jobStatus = clusterClient.getJobStatus(jobGraph.getJobID()).get(60, TimeUnit.SECONDS);
+
+		assertThat(jobStatus, isOneOf(JobStatus.CANCELED, JobStatus.CANCELLING));
+
+		final List<Path> savepoints = Files.list(savepointDirectory).map(Path::getFileName).collect(Collectors.toList());
+		assertThat(savepoints, hasItem(Paths.get(savepointLocation).getFileName()));
+	}
+
+	@Test
+	public void testDoNotCancelJobIfSavepointFails() throws Exception {
+		try {
+			Files.setPosixFilePermissions(savepointDirectory, Collections.emptySet());
+		} catch (IOException e) {
+			Assume.assumeNoException(e);
+		}
+
+		try {
+			cancelWithSavepoint();
+		} catch (Exception e) {
+			assertThat(ExceptionUtils.findThrowable(e, CheckpointTriggerException.class).isPresent(), equalTo(true));
+		}
+
+		final JobStatus jobStatus = clusterClient.getJobStatus(jobGraph.getJobID()).get(60, TimeUnit.SECONDS);
+		assertThat(jobStatus, equalTo(JobStatus.RUNNING));
+
+		// assert that checkpoints are continued to be triggered
+		triggerCheckpointLatch = new CountDownLatch(1);
+		assertThat(triggerCheckpointLatch.await(60, TimeUnit.SECONDS), equalTo(true));
+	}
+
+	private void waitForJob() throws Exception {
```

Comment:
unfortunately this is needed because the `TaskmanagerRunner` is registered in the Dispatcher after the task is started